aspnetcore/native-aot/index.md

```diff
@@ -0,0 +1,190 @@
+```
+
+The `Program.cs` source file contains some changes for publishing to .NET native AOT:
+
+```csharp
+using System.Text.Json.Serialization;
+using MyFirstAotWebApi;
+
+var builder = WebApplication.CreateSlimBuilder(args);
+builder.Logging.AddConsole();
+
+builder.Services.ConfigureHttpJsonOptions(options =>
+{
+    options.SerializerOptions.AddContext<AppJsonSerializerContext>();
+});
+
+var app = builder.Build();
+
+var sampleTodos = TodoGenerator.GenerateTodos().ToArray();
+
+var todosApi = app.MapGroup("/todos");
+todosApi.MapGet("/", () => sampleTodos);
+todosApi.MapGet("/{id}", (int id) =>
+    sampleTodos.FirstOrDefault(a => a.Id == id) is { } todo
+        ? Results.Ok(todo)
+        : Results.NotFound());
+
+app.Run();
+
+[JsonSerializable(typeof(Todo[]))]
+internal partial class AppJsonSerializerContext : JsonSerializerContext
+{
+
+}
+```
+
+A significant difference is that `Microsoft.AspNetCore.Builder.WebApplication.CreateBuilderSlim` is used to create the web application builder.  The `CreateBuilderSlim` method initializes the <xref:Microsoft.AspNetCore.Builder.WebApplicationBuilder> with incompatible ASP.NET Core features that are disabled.
+<!-- Update the preceding with the following when the .NET 8 API is published :
+<xref:Microsoft.AspNetCore.Builder.WebApplication.CreateBuilderSlim%2A>
+-->
+
+Native AOT is unable to use reflection at runtime. Source generators are used to produce code to avoid the need for reflection. In some cases source generators produce code optimized for AOT even when a generator is not strictly required. To view source code that is generated based on the code in ```Program.cs``` modify the ```MyFirstAotWebApi.csproj``` to include the ```<EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>``` property. Example:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <ServerGarbageCollection>false</ServerGarbageCollection>
+    <InvariantGlobalization>true</InvariantGlobalization>
+    <PublishAot>true</PublishAot>
+    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles> <!-- Added to emit generated files -->
+  </PropertyGroup>
+
+</Project>
+```
+
+With the project file updated, run the ```dotnet build``` command (```publish``` isn't necessary to view generated code). In the built output file there will be a ```obj/Debug/net8.0/generated/``` directory which contains all the generated files for the project.
+
+When ```dotnet publish``` is run, the project source files and generated source files are compiled as normal and then outputted assemblies are passed into an native IL linker which produces the native executable which no longer requires the .NET runtime to be installed.
+
+## Benefits of using .NET native AOT with ASP.NET Core
+
+Using the .NET native AOT deployment model provides the following benefits:
```

Comment:
Be bold and give some numbers! "Reduced startup time" is nice, but the reader will want to know much it has been reduced by.

I know numbers will move, but we can update the doc again in the future if we need to.